packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -1510,7 +1510,7 @@ function getReactiveHookCallbackIndex(calleeNode, options) {
       // useImperativeHandle(ref, fn)
       return 1;
     default:
-      if (node === calleeNode && node.name.match(/use.+Effect/)) {
+      if (node === calleeNode && node.name.match(/use.+Effect$/)) {
```

Comment:
Should we also start it with `^`?

Reply:
Thanks, Dan. Good catch.

I've added a testcase to reproduce that bug as well:

without fix
```
yarn run test ExhaustiveDeps
 yarn run v1.22.4
 $ cross-env NODE_ENV=development jest --config ./scripts/jest/config.source.j
s ExhaustiveDeps                                                             
 FAIL  packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-t
est.js                                                                       
  react-hooks
    valid
      ✕ 
function MyComponent(props) {
  return renderHelperConfusedWithEffect(() => {
    console.log(props.foo);
  }, []);
}
 (232ms)

  ● react-hooks › valid › 
function MyComponent(props) {
  return renderHelperConfusedWithEffect(() => {
    console.log(props.foo);
  }, []);
}


    assert.strictEqual(received, expected)

    Expected value to strictly be equal to:
      0
    Received:
      1
    
    Message:
      Should have no errors but had 1: [
      {
        ruleId: 'react-hooks',
        severity: 1,
        message: "React Hook renderHelperConfusedWithEffect has a missing depe
ndency: 'props.foo'. Either include it or remove the dependency array.",     
        line: 5,
        column: 6,
        nodeType: 'ArrayExpression',
        endLine: 5,
        endColumn: 8,
        suggestions: [ [Object] ]
      }
    ]

      at testValidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.j:
460:20)                                                                      
      at Object.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.j
s:662:25)                                                                    

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total
Snapshots:   0 total
Time:        1.787s, estimated 2s
Ran all test suites matching /ExhaustiveDeps/i.
 error Command failed with exit code 1.
 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this c
ommand.
```
with fix
```
$ yarn run test ExhaustiveDeps
 yarn run v1.22.4
 $ cross-env NODE_ENV=development jest --config ./scripts/jest/config.source.j
s ExhaustiveDeps                                                             
 PASS  packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-t
est.js                                                                       
  react-hooks
    valid
      ✓ 
function MyComponent(props) {
  return renderHelperConfusedWithEffect(() => {
    console.log(props.foo);
  }, []);
}
 (235ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        1.766s, estimated 2s
Ran all test suites matching /ExhaustiveDeps/i.
 Done in 2.85s.
```